diff --git a/mix.exs b/mix.exs
index 490ec4a49..155cca8c7 100644
--- a/mix.exs
+++ b/mix.exs
@@ -45,6 +45,7 @@ defmodule Ejabberd.MixProject do
                     :base64url, :fast_tls, :fast_xml, :fast_yaml, :jiffy, :jose,
                     :p1_utils, :stringprep, :syntax_tools, :yconf],
      included_applications: [:mnesia, :os_mon,
+                             :captcha, :cuesport, :fusco, :ecaptcha,
                              :cache_tab, :eimp, :mqtree, :p1_acme,
                              :p1_oauth2, :pkix, :xmpp]
      ++ cond_apps()]
@@ -100,11 +101,15 @@ defmodule Ejabberd.MixProject do
   defp deps do
     [{:base64url, "~> 1.0"},
      {:cache_tab, "~> 1.0"},
+     {:captcha, "~> 0.1", hex: :captcha_nif},
+     {:cuesport, github: "goj/cuesport", manager: :rebar3},
+     {:ecaptcha, github: "seriyps/ecaptcha"},
      {:eimp, "~> 1.0"},
      {:ex_doc, ">= 0.0.0", only: :dev},
      {:fast_tls, "~> 1.1"},
      {:fast_xml, "~> 1.1"},
      {:fast_yaml, "~> 1.0"},
+     {:fusco, "~> 0.1"},
      {:idna, "~> 6.0"},
      {:jiffy, "~> 1.1.1"},
      {:jose, "~> 1.11.5"},
