diff --git a/mix.exs b/mix.exs
index 45f2479b..532eccc1 100644
--- a/mix.exs
+++ b/mix.exs
@@ -47,6 +47,7 @@ defmodule Ejabberd.MixProject do
                           :logger, :mix]
      ++ cond_apps(),
      included_applications: [:mnesia, :os_mon,
+                             :cuesport, :fusco, :ecaptcha,
                              :cache_tab, :eimp, :mqtree, :p1_acme,
                              :p1_oauth2, :pkix]
      ++ cond_included_apps()]
@@ -131,12 +132,15 @@ defmodule Ejabberd.MixProject do

   defp deps do
     [{:cache_tab, "~> 1.0"},
+     {:cuesport, github: "goj/cuesport", manager: :rebar3},
      {:dialyxir, "~> 1.2", only: [:test], runtime: false},
+     {:ecaptcha, github: "seriyps/ecaptcha"},
      {:eimp, "~> 1.0"},
      {:ex_doc, "~> 0.31", only: [:dev, :edoc], runtime: false},
      {:fast_tls, "~> 1.1.22"},
      {:fast_xml, "~> 1.1.53", override: true},
      {:fast_yaml, "~> 1.0"},
+     {:fusco, "~> 0.1"},
      {:idna, "~> 6.0"},
      {:mqtree, "~> 1.0"},
      {:p1_acme, "~> 1.0"},
